clients/src/main/java/org/apache/kafka/common/network/NetworkReceive.java

```diff
@@ -37,20 +44,31 @@ public NetworkReceive(String source, ByteBuffer buffer) {
         this.buffer = buffer;
         this.size = null;
         this.maxSize = UNLIMITED;
+        this.memoryPool = MemoryPool.NONE;
     }
 
     public NetworkReceive(String source) {
         this.source = source;
         this.size = ByteBuffer.allocate(4);
         this.buffer = null;
         this.maxSize = UNLIMITED;
+        this.memoryPool = MemoryPool.NONE;
     }
 
     public NetworkReceive(int maxSize, String source) {
         this.source = source;
         this.size = ByteBuffer.allocate(4);
         this.buffer = null;
         this.maxSize = maxSize;
+        this.memoryPool = MemoryPool.NONE;
+    }
+
+    public NetworkReceive(int maxSize, String source, MemoryPool memoryPool) {
```

Comment:
mostly because I was going after a specific OOM scenario - DOS by large producer requests. anything can be "opted-in" to using memory pools later on, I was trying to solve just one problem.